src/core_ocean/shared/mpas_ocn_tendency.F

```diff
@@ -326,6 +336,31 @@ subroutine ocn_tend_vel(tendPool, statePool, forcingPool, diagnosticsPool, meshP
       !
       call ocn_vel_vadv_tend(meshPool, normalVelocity, layerThicknessEdge, vertAleTransportTop, tend_normalVelocity, err)
 
+      !
+      ! velocity tendency: tidal potential (if needed) 
+      !   For RK4, subtract the tidal potential from the zMid array and store in a work array, 
+      !   Then point zMid to work array so the tidal potential terms are included inside the grad computed in ocn_vel_pressure_grad_tend
+      !   zMid is pointed back to the typical value afterward, so the work array is not used in place of zMid elsewhere.
+      !
+      if (config_use_tidal_potential_forcing) then
+        call ocn_compute_tidal_potential_forcing(meshPool, forcingPool, diagnosticsPool, err)
+      end if
+
+      if (config_use_tidal_potential_forcing .and. config_time_integrator == 'RK4') then
+        call mpas_pool_get_array(forcingPool, 'tidalPotentialZMid', tidalPotentialZMid)
+        call mpas_pool_get_array(forcingPool, 'tidalPotentialEta', tidalPotentialEta)
+        call mpas_pool_get_array(meshPool, 'maxLevelCell', maxLevelCell)
+
+        do iCell = 1, nCells
+          do k = 1, maxLevelCell(iCell)
+            tidalPotentialZMid(k,iCell) = zMid(k,iCell) - tidalPotentialEta(iCell) &
+                                        - config_self_attraction_and_loading_beta * zMid(k,iCell)
```

Comment:
Will test `config_self_attraction_and_loading_beta=0` and report back because offset at `zMid` is likely cause of problem to start beside offset.

Reply:
That fixed the issue, so having a water column shifted away from 0 for `zMid` will cause a pretty spurious artificial tidal flow.

Reply:
Fixed means code was stable, but results are not correct with tides on (but are ok with tides off), which means we still have a bug here.

Reply:
# SE no wetting / drying
![8519483](https://user-images.githubusercontent.com/4295853/73078862-5752c680-3e80-11ea-829a-e8a3e2a1a4ca.png)

# RK4 no wetting / drying
![8519483](https://user-images.githubusercontent.com/4295853/73078808-3a1df800-3e80-11ea-98ac-052d04e9e46f.png)

# no tides with wetting / drying
![8519483](https://user-images.githubusercontent.com/4295853/73078755-15298500-3e80-11ea-8fac-676e806ee20e.png)

# tides with wetting / drying but no SAL
![8519483](https://user-images.githubusercontent.com/4295853/73078788-2b374580-3e80-11ea-8b43-13c607f07aaf.png)

Reply:
These are all for `USDEQU120at30cr10rr2*` meshes.

Reply:
Using the approach of adding back out the vertical shift produces:

![8537121](https://user-images.githubusercontent.com/4295853/73093457-cbe72e80-3e9b-11ea-8ca3-5a2e7ee2900a.png)

The tidal signal is contained in the output now, but the result is not as good as say just the standard RK4 without wetting / drying, so something is still not completely right.

![8537121](https://user-images.githubusercontent.com/4295853/73093811-8b3be500-3e9c-11ea-9467-8a5f2cf1e350.png)

Del4 differed between the runs, am reruning with matching del4 values.

Reply:
Note, hard-coded code change was `tidalPotentialZMid(k,iCell) = (zMid(k,iCell) + 977.502_RKIND) - tidalPotentialEta(iCell) &...`

Reply:
Station 8519483 looks better though, especially compared with the previous results.

![8519483](https://user-images.githubusercontent.com/4295853/73094437-d1457880-3e9d-11ea-9457-15e7fc7e0024.png)

Reply:
# Possible correction
@sbrus89, if we allow for zMid offset from zero, this means that this 
`- config_self_attraction_and_loading_beta * zMid(k,iCell)` may not be correct.
Shouldn't it be something more like this?
`- config_self_attraction_and_loading_beta * (ssh(iCell)-ssh_dry(iCell))`

Given it is a small correction I'm wondering if we shouldn't just turn it off.

Given the approximation really is only deep water as the text implies, maybe we should do something like this:
`- config_self_attraction_and_loading_beta * (bottomDepth > 100)*ssh(iCell)`

So, we wouldn't do any SAL for shallow waters with less than 100m depth.

# Reference
https://www.sciencedirect.com/science/article/pii/S0967064504002048
![image](https://user-images.githubusercontent.com/4295853/75492731-9ee3db00-5975-11ea-9757-4cf717fea150.png)